### PR TITLE
doc(graphql): GraphQL 101 section

### PR DESCRIPTION
**Pull request by charlypoly**
@jtoar @dthyresson 

As discussed together on Slack, here is a proposal for a "GraphQL introduction" section for the GraphQL documentation page.

### CODE REVIEW

File: docs/docs/graphql.md [hunk 1]
```diff
@@ -6,6 +6,116 @@ description: GraphQL is a fundamental part of Redwood
...
+
+# ... `User` and `Tag` type definition
+
+type Query {
+  project(name: String!): Project
+}
+```
+_More information on GraphQL types can be found [on the official GraphQL documentation](https://graphql.org/learn/schema/)_
+
+Finally, the GraphQL Schema is associated to a resolvers map that help resolving each requested fields, for example: `Project.owner`:
```
Comment (charlypoly) [inline docs/docs/graphql.md:82]:
```suggestion
Finally, the GraphQL Schema is associated with a resolvers map that helps resolve each requested field, for example: `Project.owner`:
```

**Comment by dac09** [comment]
Thanks @charlypoly - appreciate you adding this! I just added one small suggestion - I think Redwood does do graphql a little differently - in particular the "deconstructed" piece, thought it might be worth highlighting.

**Comment by charlypoly** [comment]
@dac09, you are right!
I applied your suggestions

**Comment by dthyresson** [comment]
Many thanks @charlypoly - we get many questions from developers who are new to GraphQL and this will be a great resource to point them to.

In that vein, do you think a section on "To learn more" or "Further reading" with links to other sites, blogs, documentation might help? I'm thinking of Jamie Barton's https://graphql.wtf and others.

File: docs/docs/graphql.md [hunk 2]
```diff
@@ -6,6 +6,121 @@ description: GraphQL is a fundamental part of Redwood
...
+          |                             | |    Resolvers   | |
+          |                             | |                | |
+          |                             | +--------+-------+ |
+          |     3. respond JSON data    |          |         |
+          +-----------------------------+ <--------+         |
+                                        |                    |
+                                        +--------------------+
+```
+
+For more info on a pure GraphQL implementation, [complete GraphQL guides and tutorials are available on graphql.org](https://graphql.org/learn/).
```
[the remark this answers is not in the document]
Comment (charlypoly) [reply docs/docs/graphql.md:115]: @jtoar 

The notion of "pure GraphQL implementation" was introduction by @dac09's review to distinguish the "standard" way of defining GraphQL resolvers and the way of defining resolvers in Redwood: https://github.com/redwoodjs/redwood/pull/5800#discussion_r904060217

I'm happy to change the wording if needed 🤓

**Comment by charlypoly** [comment]
@jtoar thanks for your review!

You definitely spotted all my non-native English speaker errors 😀 🦅

File: docs/docs/graphql.md [hunk 3]
```diff
@@ -1087,3 +1201,11 @@ This might be one of our most frequently asked questions of all time. Here's [To
 
 <!-- TODO -->
 <!-- This https://community.redwoodjs.com/t/how-to-add-resolvetype-resolver-for-interfaces/432/7 -->
+
+## Futher Reading
+
+Eager to learn more about GraphQL? Check out some of the resources below:
+
+- [GraphQL.wtf](https://graphql.wtf) covers most aspects of GraphQL and publishes one short video a week 
+- [howtographql.com] is aimed at people looking to learn GraphQL in a practical way, through a set of complete full-stack tutorials
```
Comment (Urigo) [inline docs/docs/graphql.md:1210]: it's not so up to date tutorial...
Should maybe change that to the Yoga tutorial?
It's more up to date and also is the thing that Redwood actually use under the hood?

Comment (charlypoly) [reply docs/docs/graphql.md]: @Urigo Fixed!

**Review by jtoar — APPROVED** [review]
Thanks for your patience @charlypoly, let's get this merged!

**Pull request merged**